src/dependency-manager/spec/transform/component-transform.ts

```diff
@@ -55,6 +56,12 @@ export const transformComponentSpec = (spec: ComponentSpec): ComponentConfig =>
   const services = transformDictionary(transformServiceSpec, spec.services, spec.metadata);
   const tasks = transformDictionary(transformTaskSpec, spec.tasks, spec.metadata);
   const dependencies = spec.dependencies || {};
+  const service_databases = transformDictionary(transformDatabaseSpecToServiceSpec, spec.databases, spec.metadata);
+  const databases = transformDictionary(transformDatabaseSpec, spec.databases, spec.metadata);
+  for (const [key, value] of Object.entries(service_databases)) {
+    service_databases[`${key}-db`] = value;
```

Comment:
The problem is that `--` is not valid as a name in our system. So after we generate the service our validation throws an error. This should only be a problem for a bit until we add in support for fully managed databases and then this will go away since we will have code that does not treat databases as services.